dandi/validate.py

```diff
@@ -33,9 +45,38 @@ def validate_bids(
     """
     from .bids_validator_xs import validate_bids as validate_bids_
 
-    return validate_bids_(
+    validation_result = validate_bids_(
         paths, schema_version=schema_version, debug=devel_debug, report_path=report
     )
+    missing_files = [
+        pattern["regex"]
+        for pattern in validation_result["schema_tracking"]
+        if pattern["mandatory"]
+    ]
+    error_list = []
+    if missing_files:
+        error_substring = (
+            f"{pluralize(len(missing_files), 'filename pattern')} required "
+            "by BIDS could not be found"
+        )
+        error_list.append(error_substring)
+    if validation_result["path_tracking"]:
+        error_substring = (
+            f"{pluralize(len(validation_result['path_tracking']), 'filename')} "
+            "did not match any pattern known to BIDS"
+        )
+        error_list.append(error_substring)
+    if error_list:
+        error_string = " and ".join(error_list)
+        error_string = f"Summary: {error_string}."
+        click.secho(
```

Comment:
I don't like this.  Errors in a library function should be reported via a return value or exception for printing by the caller.

@yarikoptic Thoughts on how BIDS dataset validation errors should be displayed when uploading?

Reply:
This I am unsure about, as it stands the main “library function” for validation (`bids_validator_xs.validate_bids()`) just returns a nested dictionary which we could build our reporting on. If we don't want to have this `click` call happen in `validate.py` we would need to reproduce this in both `upload.py` and `cli/cmd_validate.py` — because both modules use validation and (I woud assume?) need identical reporting to not confuse people when they get one sort of output for validation, and another sort when they try to upload.

Reply:
Agree with @jwodder that there should be no `print` or `click.anything` in the library, non-CLI code. Details should be logged (.error or .warning), and display/reaction during upload should be exactly the same  as for nwb files. Check details, but IIRC we would just error out that file upload unless --validation is `skip` or `ignore`.  Overall: just mimic behavior on what is happening for NWB files ATM.  Let me know if I should analyze and provide more specific "desires".